content/doc/book/pipeline/shared-libraries.adoc

```diff
@@ -218,204 +218,173 @@ this variable to checkout the appropriate version of the library.
 
 image::pipeline/global-pipeline-library-legacy-scm.png["Configuring a 'Legacy SCM' for a Pipeline Library", role=center]
 
+=== Pretesting library changes
 
-== Writing libraries
+If you notice a mistake in a build using an untrusted library,
+simply click the _Replay_ link to try editing one or more of its source files,
+and see if the resulting build behaves as expected.
+Once you are satisfied with the result, follow the diff link from the build’s status page,
+and apply the diff to the library repository and commit.
 
-At the base level, any valid
-link:http://groovy-lang.org/syntax.html[Groovy code]
-is okay for use. Different data structures, utility methods, etc, such as:
+(Even if the version requested for the library was a branch, rather than a fixed version like a tag,
+replayed builds will use the exact same revision as the original build:
+library sources will not be checked out again.)
 
-[source,groovy]
-----
-// src/org/foo/Point.groovy
-package org.foo;
+_Replay_ is not currently supported for trusted libraries.
+Modifying resource files is also not currently supported during _Replay_.
 
-// point in 3D space
-class Point {
-  float x,y,z;
-}
-----
+== Writing and using library contents
 
-=== Accessing steps
+Now let's look at how to add items to Shared Libraries and how to use those items.
 
-Library classes cannot directly call steps such as `sh` or `git`.
-They can however implement methods, outside of the scope of an enclosing
-class, which in turn invoke Pipeline steps, for example:
+=== Resources
+Shared libraries can contain general resource files, such as properties files.
+To add resources, simply add files to the `resources/` directory tree.
+External libraries may then load these files using the `libraryResource` step,
+which accepts a relative pathname, akin to Java resource loading:
 
 [source,groovy]
 ----
-// src/org/foo/Zot.groovy
-package org.foo;
-
-def checkOutFrom(repo) {
-  git url: "git@github.com:jenkinsci/${repo}"
-}
+def request = libraryResource 'com/mycorp/pipeline/somelib/request.json'
 ----
 
-Which can then be called from a Scripted Pipeline:
+The file is loaded as a string, suitable for passing to APIs or saving
+to a workspace using `writeFile`.
 
-[source,groovy]
-----
-def z = new org.foo.Zot()
-z.checkOutFrom(repo)
-----
+Avoid adding resources directly under the `resources/` directory.
+Multiple libraries maybe be loaded by one pipeline, but resources are referenced using
+only a relative path.
+Use a deeper, unique directory structure to avoid resource name conflicts with other libraries.
 
-This approach has limitations; for example, it prevents the declaration of a
-superclass.
+=== Global variables
 
-Alternately, a set of `steps` can be passed explicitly to a library class, in a
-constructor, or just one method:
+If a file in the `vars` directory contains one or more methods (other than `call`),
+a global variable of that name is created, and the methods are added to that global.
```

Comment:
Just a minor enhancement to make it even clearer: "[...] a global variable of that [file] name is created [...]" or should it be "[...] a global variable [with] that [file] name is created [...]"?